### ISSUE TITLE
Creating new app fails/hangs

### ISSUE THREAD

**Issue by nilay-automatesmb** [apache/incubator-predictionio#253]
Have tried creating new app on various servers like from Digitalocean, Google Compute cloud all with more than 4GB of memory. But pio app new <app name> always hangs. 

Upon force exit, and doing pio app list show 1 app but there is no ID,Access Key

If we try to create a new app with same name it throws error that app already exist.

Have waited more than 3 hours,, but no result.

Though this works fine when tried on local system.

**Comment by Nigeltiany** [comment]
+1 

Happened to me too. Very Frustrating. I suspect its the location/placement of the folder which holds engine template. I suspect its because i had it deeper in PredictionIO\bin file. So maybe the process didn't have the required files the complete the who operation(dead lock). But this is just speculation. After the installation process not being as simple as i had hoped, i didn't investigate further. I had to install PredictionIO by building it from source because the install script doesn't work and after the hustle i still didn't expose the api or do any actual learning with the api.

**Comment by dszeto** [comment]
For the curious, the root cause of this is due to the implementation of access key generation (https://github.com/apache/incubator-predictionio/blob/develop/data/src/main/scala/org/apache/predictionio/data/storage/AccessKeys.scala#L68) and the lack of entropy on virtual machines (whether locally or on the cloud) explained here: https://blog.cloudflare.com/ensuring-randomness-with-linuxs-random-number-generator/

tl;dr, if you are just testing out PredictionIO and are running into this problem, please specify your own access key: `pio app new your_app_name --access-key your_key` to workaround this problem.

We are looking to a way of adding a command line argument to unblock progress at the cost of using a less secure random source.

**Comment by jsyrjala** [comment]
I was hit by this also. I was running prediction-io in Amazon, and there executing `pio new app` took around 10 minutes. Its not hanging, it just takes a very long time to finish.

Using `--access-key` fixes this.

**Comment by jsyrjala** [comment]
Related ticket in Apache JIRA https://issues.apache.org/jira/browse/PIO-59

**Comment by jsyrjala** [comment]
Pull request https://github.com/apache/incubator-predictionio/pull/367

**Comment by AnaMakarevich** [comment]
So did it work? I tried changing that file with AccessKey generation but without any result. I mean - it still freezes.

**Comment by PanJ** [comment]
My workaround without using `--access-key` is using `rdgd` with `/dev/urandom` as a seed by running `rngd -r /dev/urandom -o /dev/random`. Any key generation after this will be fast.
